lambdas/delete_appointment_block.py

```diff
@@ -35,26 +35,33 @@ def lambda_handler(event, context):
     else:
         raise LambdaException("Invalid input: No appointment Id")
 
-    #Check if appointment exists
+    #Check if appointment_block_id exists in appointment_block
     sql = "SELECT appointment_block_id FROM appointment_block WHERE supporter_id= :supporter_id"
     supporter_id_param = [{'name' : 'supporter_id', 'value' : {'longValue' : supporter_id}}]
     existing_appointment_block = query(sql, supporter_id_param)
 
     if(existing_appointment_block['records'] == []):
         print("No existing appointment block")
         raise LambdaException("No existing appointment block")
-        
-    sql = """DELETE FROM specializations_for_block WHERE appointment_block_id= :appnt_blck_id"""
-    sql_parameters = [{'name' : 'appnt_blck_id', 'value': {'longValue' : appointment_block_id}}] 
 
-    delete_appmnt_blck = query(sql,sql_parameters)
+    #Check if appointment_block_id exists in specializations_for_block
+    sql = "SELECT appointment_block_id FROM specializations_for_block WHERE appointment_block_id= :appointment_block_id"
+    supporter_id_param = [{'name' : 'appointment_block_id', 'value' : {'longValue' : appointment_block_id}}]
+    existing_appointment_block = query(sql, supporter_id_param)
 
-    # check if delete successfull
-    if delete_appmnt_blck['numberOfRecordsUpdated'] == 0:
-        return {
-            'body': json.dumps("appointment block not deleted from specializations_for_block"),
-            'statusCode': 404
-        }
+    if(existing_appointment_block['records'] == []):
+        print("No existing specializations for appointment block in specializations_for_block")
+    else:
+        sql = """DELETE FROM specializations_for_block WHERE appointment_block_id= :appnt_blck_id"""
+        sql_parameters = [{'name' : 'appnt_blck_id', 'value': {'longValue' : appointment_block_id}}] 
+        delete_appmnt_blck = query(sql,sql_parameters)
+        
+        # check if delete successfull
+        if delete_appmnt_blck['numberOfRecordsUpdated'] == 0:
+            return {
+                'body': json.dumps("appointment block not deleted from specializations_for_block"),
+                'statusCode': 404
```

Comment:
I believe you need to throw a lambda exception here as well instead of returning error code in the body.